### ISSUE TITLE
`Fix` and `Vary` missing `AnnoFloat` layer in parsing

### ISSUE THREAD

**Issue by hypotext** [penrose/penrose#444]
Repro and example: same as #443 

Expected: values tagged `Fix` occur inside something tagged `AnnoFloat` in an `Expr`

Got: `Fix` occurs directly inside `Expr`:

```
                        "expr": {
                            "start": {
                                "line": 6,
                                "col": 27
                            },
                            "end": {
                                "line": 6,
                                "col": 32
                            },
                            "tag": "Fix",
                            "contents": 350
                        }
```

Suggestion: Change parser to nest `Fix` inside `AnnoFloat`

This also seems to be the case for `Vary`.

**Comment by hypotext** [comment]
Updated to reflect that the same error happens for `Vary`

**Comment by wodeni** [comment]
Not sure what your use case will be. Two possible solutions: (1) directly put `AnnoFloat` instead of `IAFloat` into `Expr`, which effectively gets rid of the extra nesting; (2) your proposal.

**Comment by wodeni** [comment]
Fixed in c952679 (solution 1). I don't see an obvious reason for the extra nesting. I imagine you'll just separately match on `Vary` and `Fix` tags. The sum type `AnnoFloat` can still be used to type values of either types.
